sqlite3.go

```diff
@@ -719,93 +799,142 @@ func (c *SQLiteConn) RegisterAggregator(name string, impl any, pure bool) error
 	}
 
 	agg := t.Out(0)
+	var implReturnsPointer bool
 	switch agg.Kind() {
-	case reflect.Ptr, reflect.Interface:
+	case reflect.Ptr:
+		implReturnsPointer = true
+	case reflect.Interface:
+		implReturnsPointer = false
 	default:
-		return errors.New("SQlite aggregator constructor must return a pointer object")
+		return errors.New("SQLite aggregator constructor must return a pointer object")
 	}
+
 	stepFn, found := agg.MethodByName("Step")
 	if !found {
-		return errors.New("SQlite aggregator doesn't have a Step() function")
+		return errors.New("SQLite aggregator doesn't have a Step() function")
+	}
+	err := ai.setupStepInterface(stepFn, &ai.stepArgConverters, &ai.stepVariadicConverter, implReturnsPointer, "Step()")
```

Comment:
The penultimate parameter to `setupStepInterface` is named `isImplPointer`. That doesn't seem to have anything to do with `implReturnsPointer`.

Reply:
I could rename the parameter to `hasMethodReceiver`. 

I maintained the behavior from the original implementation, where we'd only skip the method receiver if the `agg.Kind() == reflect.Pointer`, but if we must always return a pointer, then there should always be a method receiver. 

According to the [`reflect.Type` docs](https://pkg.go.dev/reflect#Type):

>	// For a non-interface type T or *T, the returned Method's Type and Func
>	// fields describe a function whose first argument is the receiver,
>	// and only exported methods are accessible.


When running `sqlite3_test.go`, I did not encounter any cases where `implReturnsPointer` would be false. I wonder if we should remove this branch of the logic.